e2e-tests/playwright/docs/accessibility/interaction_testing.md

```diff
@@ -0,0 +1,405 @@
+    await button.blur();
+    await button.click();
+    await expect(button).toBeFocused();
+});
+```
+
+### **2. Complex Navigation Patterns**
+
+```typescript
+test('hierarchical menu navigation', async ({page}) => {
+    await page.locator('[data-testid="menu-button"]').focus();
+    await page.keyboard.press('Enter');
+
+    // Navigate to submenu trigger
+    const submenuTrigger = page.getByRole('menuitem', {name: /has submenu/});
+    await submenuTrigger.focus();
+
+    // Open submenu with right arrow
+    await page.keyboard.press('ArrowRight');
+
+    // Verify submenu opened and focused
+    const submenu = page.locator('[role="menu"]').nth(1);
+    await expect(submenu).toBeVisible();
+    await expect(submenu.getByRole('menuitem').first()).toBeFocused();
+
+    // Navigate in submenu
+    await page.keyboard.press('ArrowDown');
+    await expect(submenu.getByRole('menuitem').nth(1)).toBeFocused();
+
+    // Return to parent menu with left arrow
+    await page.keyboard.press('ArrowLeft');
+    await expect(submenu).toBeHidden();
+    await expect(submenuTrigger).toBeFocused();
+});
+```
+
+### **3. Screen Reader Simulation**
+
+```typescript
+test('screen reader content structure', async ({page}) => {
+    // Test heading structure
+    const headings = page.locator('h1, h2, h3, h4, h5, h6');
+    const headingLevels = await headings.evaluateAll((elements) =>
+        elements.map((el) => parseInt(el.tagName.substring(1))),
+    );
+
+    // Verify logical heading hierarchy
+    for (let i = 1; i < headingLevels.length; i++) {
+        const levelDiff = headingLevels[i] - headingLevels[i - 1];
+        expect(levelDiff, 'Heading levels should not skip').toBeLessThanOrEqual(1);
+    }
+
+    // Test landmark navigation
+    const landmarks = page.locator('[role="banner"], [role="navigation"], [role="main"], [role="contentinfo"]');
+    const landmarkCount = await landmarks.count();
+    expect(landmarkCount, 'Page should have proper landmarks').toBeGreaterThan(0);
+
+    // Test skip links
+    await page.keyboard.press('Tab');
+    const firstFocusable = await page.evaluate(() => document.activeElement?.textContent);
+    expect(firstFocusable, 'First focusable should be skip link').toContain('Skip');
+});
+```
+
+## Utility Functions
+
+### **Focus Management Helpers**
+
+```typescript
+// Test focus restoration after modal close
+async function testFocusRestoration(page: Page, triggerSelector: string, modalSelector: string) {
+    const trigger = page.locator(triggerSelector);
+
+    // Open modal
+    await trigger.click();
+    await expect(page.locator(modalSelector)).toBeVisible();
+
+    // Close modal with Escape
+    await page.keyboard.press('Escape');
+    await expect(page.locator(modalSelector)).toBeHidden();
+
+    // Verify focus returned
+    await expect(trigger).toBeFocused();
+}
+```
+
+## Performance Considerations
+
+### **Efficient Focus Testing**
```

Comment:
Do we actually care that much about performance for these tests? Also, it's just testing that the browser is handling `:focus-visible` again